packages/settings-library/src/settings_library/node_ports.py

```diff
@@ -0,0 +1,46 @@
+from typing import ClassVar, Final
+
+from pydantic import Field, NonNegativeInt, PositiveInt, SecretStr
+
+from ._constants import MINUTE
+from .base import BaseCustomSettings
+from .basic_types import PortInt
+from .postgres import PostgresSettings
+from .storage import StorageSettings
+
+NODE_PORTS_400_REQUEST_TIMEOUT_ATTEMPTS_DEFAULT_VALUE: Final[NonNegativeInt] = 3
+
+
+class StorageAuthSettings(BaseCustomSettings):
+    NODE_PORTS_STORAGE_HOST: str
+    NODE_PORTS_STORAGE_PORT: PortInt
+
+    NODE_PORTS_STORAGE_LOGIN: str
+    NODE_PORTS_STORAGE_PASSWORD: SecretStr
+
+    @property
+    def base_url(self) -> str:
+        return StorageSettings(
+            STORAGE_HOST=self.NODE_PORTS_STORAGE_HOST,
+            STORAGE_PORT=self.NODE_PORTS_STORAGE_PORT,
+        ).base_url
+
+    class Config:
+        json_encoders: ClassVar = {
+            SecretStr: lambda v: v.get_secret_value() if v else None,
```

Comment:
IMO this is a bad idea since it invalidates using `SecretStr`. 
Anywhere in the code calling `settings.json()` in e.g. a log would leak the secrets
I advice to do it dynamically and not leave it as a "static" config.

Reply:
Good point. I have added custom methods for this